src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -197,12 +361,22 @@ protected override IEnumerable<AzureSqlManagedInstanceModel> ApplyUserInputToMod
             AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
             Management.Internal.Resources.Models.Sku Sku = new Management.Internal.Resources.Models.Sku();
 
-            if (Edition != null)
+            // Get current edition
+            string currentEdition = existingInstance.Sku.Tier;
+
+            // Get current hardware family
+            string currentComputeGeneration = existingInstance.Sku.Name.Contains(Constants.ComputeGenerationGen4) ? Constants.ComputeGenerationGen4 : Constants.ComputeGenerationGen5;
+
+            if (SkuName != null)
             {
-                string computeGeneration = existingInstance.Sku.Name.Contains(Constants.ComputeGenerationGen4) ? Constants.ComputeGenerationGen4 : Constants.ComputeGenerationGen5;
-                string editionShort = AzureSqlManagedInstanceAdapter.GetInstanceSkuPrefix(Edition);
-                Sku.Name = editionShort + "_" + computeGeneration;
-                Sku.Tier = Edition;
+                Sku.Name = SkuName;
```

Comment:
Why not populating Sku.Tier and Sku.Family here?

Reply:
I've omitted for brevity because Sku.Name is enough for upsert

Reply:
If you need to explain in PR, it's worth adding a code comment :)

Reply:
Nice tip. Thanks for that. Adding comment